Basic-Car-Maintenance/Shared/Models/AppIcon.swift

```diff
@@ -0,0 +1,45 @@
+import UIKit
+
+enum AppIcon: String, CaseIterable, Identifiable {
+  case primary = "AppIcon"
+  case carDark = "AppIcon-car-dark"
+  case carRed = "AppIcon-car-red"
+  case carYellow = "AppIcon-car-yellow"
+  case carBlack = "AppIcon-car-black"
+  case carOrange = "AppIcon-car-orange"
+  
+  var id: String { rawValue }
+  
+  /// the name of the icon in the App Bundle.
+  var iconName: String? {
+    switch self {
+    /// returns `nil`. Use this case to reset the app icon back to its primary icon.
+    case .primary:
+      return nil
+    default:
+      return rawValue
+    }
+  }
+  
+  /// A UI presentable string.
+  var description: String {
+    switch self {
+    case .primary:
+      return "Default"
+    case .carDark:
+      return "Dark Mode"
+    case .carRed:
+      return "Red Car"
+    case .carYellow:
+      return "Yellow Car"
+    case .carBlack:
+      return "Black Car"
+    case .carOrange:
+      return "Orange Car"
+    }
+  }
+  
+  var preview: UIImage {
+    UIImage(named: rawValue) ?? UIImage()
+  }
```

Comment:
Is there a specific reason this has to be a `UIImage` rather than an `Image`?

Reply:
Because the image is stored in a .appIconSet, the images did not load using a plain old SwiftUI String initialization like `Image(icon.rawValue)`. (See  ChooseAppIconView (line 64).) All of the implementations that I saw so far, including the SwiftLee article you sent (which was very helpful by the way 👍🏼) used the UIImage method. 

So far I haven't found a purely SwiftUI way to do this. I could do more research on it, but I'm not sure that there's a major benefit to it.

Reply:
all good, I was curious!